### ISSUE TITLE
AND condition for filter on a single column

### ISSUE THREAD

**Issue by Alfagun74** [ppetzold/nestjs-paginate#402]
I wanted to match mutliple genres of a game with an **and** condition, but only the last condition gets applied. How to achieve something like this?  
  
`&filter.genres.name=$eq:Indie&filter.genres.name=$eq:RPG`  

`&filter.genres.name=$eq:Indie,RPG` -> wont work  
  
`&filter.genres.name=$in:Indie,RPG` -> is OR condition

**Comment by ppetzold** [comment]
I think, the first one should be working the way the helper function is written. However, the query parser won't allow it.

https://github.com/ppetzold/nestjs-paginate/blob/master/src/decorator.ts#L53

For `&filter.genres.name=$eq:Indie&filter.genres.name=$eq:RPG`

It will always parse one statement per column:

`{filter: { 'genres.name': $eq:RPG }}`

But right, this limits conditions for 1:M and N:M. We could adapt the following format for parsing the filter conditions:

`{filter: [{ 'genres.name': $eq:Indie }, { 'genres.name': $eq:RPG }]}`

So AND conditions can be handled.

Would accept a PR with this breaking change :)

**Comment by ppetzold** [comment]
Would generally improve the possibilities of conditions one could do on a column.

`?filter.name=$like:foo&filter.name=$not:foobar`

**Comment by ppetzold** [comment]
should solve #77 as well

**Comment by ppetzold** [comment]
Thinking about it. This format seems like a better extension:

`{filter: { 'genres.name': ['$eq:Indie, '$eq:RPG'] }}`

than

`{filter: [{ 'genres.name': $eq:Indie }, { 'genres.name': $eq:RPG }]}`

**Comment by ppetzold** [comment]
Closed #457

**Comment by Alfagun74** [comment]
Hey @ppetzold,

I'm having trouble with the query string `&filter.tags.name=$eq:2D&filter.tags.name=$and:$eq:Singleplayer`. The issue seems to stem from applying the equality operator `$eq` separately to each tag name. The 'tags' field is an array of objects, and the current method doesn't give the results I want.

The SQL query it generates is trying to find games with both tags at the same time using the equality operator, resulting in an empty dataset. I'd prefer a functionality like `$contains` for more complex arrays or a solution allowing the use of 'AND' conditions with these specific tags.

Here's the SQL example it generates:

```sql
WHERE ( 1=1 AND ("__root_tags_rel"."name" = ? AND "__root_tags_rel"."name" = ?) ) AND ( "__root"."deleted_at" IS NULL ) -- PARAMETERS: ["2D","Singleplayer"]
```

The tags are stored as objects with associated data, like:

```
Game -> Tags -> Tag 1 -> name: Singleplayer <- filter
                      relatedGames
                      id
                      etc
             -> Tag 2 -> name: 2D <- filter
                      relatedGames
                      id
                      etc
```

I want to find games with both 'Singleplayer' and '2D' tags. While the $in operator helps find games with either tag, it doesn't meet the requirement of finding games **only** with both tags.

Is there a way to achieve this or a workaround allowing the use of 'AND' conditions for these specific tags or using $contains for more complicated arrays?
